core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunner.java

```diff
@@ -66,21 +66,25 @@ public Result build(List<String> args, String nativeImageName, String resultingE
             int exitCode = process.waitFor();
             boolean objcopyExists = objcopyExists();
             if (exitCode != 0) {
-                return new Result(exitCode, objcopyExists);
+                return new Result(exitCode);
             }
 
             if (objcopyExists) {
-                if (debugSymbolsEnabled) {
+                if (debugSymbolsEnabled && graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) < 0) {
+                    // Need to explicitly split debug symbols prior to GraalVM/Mandrel 23.0
                     splitDebugSymbols(nativeImageName, resultingExecutableName);
                 }
-                // Strip debug symbols regardless, because the underlying JDK might contain them
-                objcopy("--strip-debug", resultingExecutableName);
+                if (!debugSymbolsEnabled || graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) < 0) {
+                    // Strip debug symbols even if not generated by GraalVM/Mandrel, because the underlying JDK might
+                    // contain them
+                    objcopy("--strip-debug", resultingExecutableName);
+                }
             } else if (SystemUtils.IS_OS_LINUX) {
```

Comment:
So on a system without `objcopy` and GraalVM  `>= 23.0` and debuginfo turned on, we'd get duplicate warnings about `objcopy` missing. Once by quarkus a second time from `native-image`.